src/OrchardCore.Modules/Orchard.Autoroute/Services/AutorouteValidator.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Extensions.Localization;
+using Orchard.Autoroute.Model;
+using Orchard.ContentManagement.Records;
+using System;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using YesSql.Core.Services;
+
+namespace Orchard.Autoroute.Services
+{
+    public class AutorouteValidator : IAutorouteValidator
```

Comment:
If you really need to externalize the validation logic, you can just create another driver, it doesn't need its own service.